core/settings/qPrefDisplay.h

```diff
@@ -22,11 +23,11 @@ class qPrefDisplay : public QObject {
 	void sync() { loadSync(true); }
 
 public:
-	const QString divelist_font() const;
-	double font_size() const;
-	bool display_invalid_dives() const;
-	bool show_developer() const;
-	const QString theme() const;
+	static inline const QString divelist_font() {return QString(prefs.divelist_font); };
+	static inline double font_size() {return prefs.font_size; };
+	static inline bool display_invalid_dives() {return prefs.display_invalid_dives; };
+	static inline bool show_developer() {return prefs.show_developer; };
+	static inline const QString theme() {return QString(prefs.theme); };
```

Comment:
so, just to make sure I understand this correctly - the `inline` is redundant but a reminder to the reader that the compiler may decide to inline this. And the fact that we want the compiler to inline this is why the implementation is in the `.h` instead of the `.cpp`, correct?

Reply:
1) If you want it inlined *and* used from different parts of the codebase, you should obviously put it in a header file, yes.
2) There are two ways of making a member function inline. Firstly, the common way, outside of the class definition:
```
class {
    int f();
};
[...]
inline int f()
{
    return -1;
}
```
I think this is the cleaner approach, as it separates function declaration from function definition. Some people collect the inlined functions and templates in a separate "*_impl.h" header. Inlining is an implementation detail. Why would the user of the function care if it's inline? That decision should be up to the writer of the function.
But you can also do it by the function-definition in class-definition method:
```
class {
    int f() { return -1; }
};
```
This is implicitly inline. It *has* to be, because otherwise the program couldn't link if you use the class in different translations units. The advantage is less lines of code. @janiversen had a mess of sometimes with and sometimes without the keyword. Unfortunately, he interpreted my comment as a request to add it everywhere (story of my life). Still that's better than inconsistency.
3) I find the whole inlining practice questionable once `QString` is involved. I tried it and `gcc7` is surprisingly smart about *not* inlining too much. For example the `strlen` is not inlined for me. So it really depends on in how many places these functions are called. If they are called in one or no places, inlining is of course a win. If they are called numerous times, it is a loss. In the end, we probably should rely on the compiler to do the right thing.